lib/addons/karpenter/index.ts

```diff
@@ -469,6 +469,10 @@ export class KarpenterAddOn extends HelmAddOn {
 
         karpenterChart.node.addDependency(ns);
 
+        if(clusterInfo.nodeGroups) {
```

Comment:
@youngjeong46 this is what I added to fix. I was able to run it once e2e in the target account, I will run one more time. 
This code should only refence the node group (or fargate profile) used to deploy karpenter. I added deps on all node groups.